### PR TITLE
fix: prevent extending if template disallows

### LINKED ISSUE #13078: Template-enforced autostop can be overridden in the UI
Users can override the default template Autostop in the UI even if it's prevented in the template settings.

autostop_enforcement

stop_in_13_hours

Tested with both admin and user roles, though I'm not sure if it can be modified via CLI/API from the user.

Effectively makes the "enforce changes..." option obsolete. 

---

Two options for fixing this: 

We can disable [+] and leave [-] enabled, since we don't want to prevent users from reducing cloud spend.

Or we can hide the [-] [+] entirely for simplicity.

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Closes https://github.com/coder/coder/issues/13078

### CODE REVIEW

**Comment by BrunoQuaresma** [comment]
@stirby I would appreciate your QA on this PR since auto stop has been a sensitive feature.

File: coderd/workspaces.go [hunk 1]
```diff
@@ -1052,6 +1052,18 @@ func (api *API) putExtendWorkspace(rw http.ResponseWriter, r *http.Request) {
...
+		tmpl, err := s.GetTemplateByID(ctx, workspace.TemplateID)
+		if err != nil {
+			code = http.StatusInternalServerError
+			resp.Message = "Error fetching template."
+			return xerrors.Errorf("get template: %w", err)
+		}
+		if !tmpl.AllowUserAutostop {
+			code = http.StatusBadRequest
+			resp.Message = "Cannot extend workspace: template does not allow user autostop."
+			return xerrors.New("Cannot extend workspace: template does not allow user autostop")
```
Comment (mafredri) [inline coderd/workspaces.go:1064]:
```suggestion
			return xerrors.New("cannot extend workspace: template does not allow user autostop")
```

**Review by mafredri — APPROVED** [review]
LGTM!

**Review by aslilac — APPROVED** [review]

**Review by mtojek — APPROVED** [review]
Tiny flake in `TestWorkspaceProxyWorkspaceApps_BlockDirect/PortSharing/HTTPS` but unrelated 👍

**Comment by stirby** [comment]
@BrunoQuaresma this is great - simple and clean solution. Validated the behavior in preview deployment, nice work.

**Review by stirby — APPROVED** [review]

**Pull request merged**